### PR TITLE
[Py] Python wrappers for nGraph ops.

### PR DESCRIPTION
**Pull request by arogowie-intel**
- AvgPool
- Broadcast
- Ceiling
- Constant
- Convert
- Convolution

### CODE REVIEW

File: python/ngraph/utils/broadcasting.py [hunk 1]
```diff
@@ -34,15 +34,15 @@ def get_broadcast_axes(left_shape, right_shape, axis):
     For example, an output shape of `{2,5,6,2,8}` and input shape of `{2,6}` means
     that the broadcast axes must be `{1,3,4}`.
     """
-    axes_indexes = list(range(0, len(left_shape)))
+    axes_indexes = list(range(0, len(output_shape)))
     if axis is None:
-        right_begin = len(left_shape) - len(right_shape)
```
Comment (postrational) [inline python/ngraph/utils/broadcasting.py]: If we're changing names from left, right to in, out let's be consistent and rename all variables.

**Comment by arogowie-intel** [comment]
As for `get_broadcast_axes` function docstring - I'm not sure how `axis` parameter works thus I omitted it's description. I don't know how exactly nGraph `Broadcast` operation works and since I can't determine if this whole function is correct.

File: python/ngraph/utils/broadcasting.py [hunk 2]
```diff
@@ -14,35 +14,38 @@
...
+        output_begin = len(output_shape) - len(input_shape)
     else:
-        right_begin = axis
-    right_axes_indexes = list(range(right_begin, right_begin + len(right_shape)))
+        output_begin = axis
+    right_axes_indexes = list(range(output_begin, output_begin + len(input_shape)))
     for index in reversed(right_axes_indexes):
         del axes_indexes[index]
-    return AxisSet(set(axes_indexes))
+    return set(axes_indexes)
```
Comment (tsocha) [inline python/ngraph/utils/broadcasting.py]: Why the return type was changed?

Comment (postrational) [reply python/ngraph/utils/broadcasting.py]: +1

File: python/ngraph/ops.py [hunk 3]
```diff
@@ -274,49 +282,77 @@ def dot(left_node, right_node, name=None):
 
 # convpool ops
 @nameable_op
-def convolution(x,                      # type: Node
```
Comment (postrational) [inline python/ngraph/ops.py]: Let's verify that names here are identical (or similar) to ngraph++ API names.

Comment (arogowie-intel) [reply python/ngraph/ops.py]: Updated.

File: python/test/ngraph/test_reduction.py [hunk 4]
```diff
@@ -17,9 +17,10 @@
 import pytest
 
 import ngraph as ng
+from test.ngraph.util import run_op_node
 
 
-@pytest.mark.parametrize('ng_api_helper, numpy_function, reduction_axes', [
+@pytest.mark.parametrize('ng_fun, numpy_function, reduction_axes', [
```
Comment (postrational) [inline python/test/ngraph/test_reduction.py]: `ng_api_helper` was more descriptive then `ng_fun`

File: python/test/ngraph/util.py [hunk 5]
```diff
@@ -13,32 +13,28 @@
...
     manager_name = pytest.config.getoption('backend', default='CPU')
-    rt = ng.runtime(manager_name=manager_name)
+    return ng.runtime(manager_name=manager_name)
 
-    data = np.arange(11, 27, dtype=np.float32)
-    data = data.reshape((1, 1, 4, 4))
-    data = np.broadcast_to(data, (1, 1, 4, 4, 4))
 
-    param = ng.parameter(data.shape)
+def run_op_node(input_data, op_fun, *args):
```
Comment (postrational) [inline python/test/ngraph/util.py]: `op_fun` is not obvious. Change name or document.

File: python/test/ngraph/test_pooling.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+from test.ngraph.util import get_runtime
+
+
+@pytest.fixture
+def _ndarray_1x1x4x4():
+    return np.arange(11, 27, dtype=np.float32).reshape(1, 1, 4, 4)
+
+
+def test_avg_pool_2d(_ndarray_1x1x4x4):
+    rt = get_runtime()
```
Comment (postrational) [inline python/test/ngraph/test_pooling.py]: Don't use abbreviated variable names

**Comment by postrational** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by DawnStone** [comment]
The build failed due to a style check:
```check-code-format.sh STATUS: About to format C/C++ code in directory tree '/home/dockuser/ngraph/python/pyngraph' ...
~/ngraph/BUILD-CLANG/test
1 of 657 source files failed the code-format check:
    python/pyngraph/ops/constant.cpp
test/CMakeFiles/style-check.dir/build.make:57: recipe for target 'test/CMakeFiles/style-check' failed
make[3]: *** [test/CMakeFiles/style-check] Error 1
```

**Comment by raramer01** [comment]
TESTNOW

**Comment by raramer01** [comment]
the python_ci check encounters the serializer compilation problem that is due to a change that was made to the serializer api, see PR #780 

maybe, need to do another merge master into the branch before doing another TESTNOW...

**Comment by arogowie-intel** [comment]
Please merge first #780 and then this PR.

**Pull request merged**